ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -117,6 +124,16 @@ public void init() {
           try {
             CqlRequestImpl cqlRequest = GSON.fromJson(util.safeGetBody(req), CqlRequestImpl.class);
             CqlQueryResponse cqlQueryResponse = cqlQueryUtil.executeCqlQuery(cqlRequest);
+            if (cqlQueryResponse.getQueryResponse() != null) {
+              cqlQueryResponse
+                  .getQueryResponse()
+                  .getProcessingDetails()
+                  .stream()
+                  .map(sourceWarningsFilterManager::filterWarningsOf)
+                  .map(ProcessingDetails::getWarnings)
+                  .filter(Objects::nonNull)
+                  .forEach(cqlQueryResponse.getWarnings()::addAll);
```

Comment:
Can we guarantee that each implementation of `CqlQueryResponse.getWarnings()` never returns null? I guess I could add a comment like `// must not return null` to the method and then make sure that each implementation has a test which verifies that `getWarnings()` never returns null. Alternatively, I could verify that `cqlQueryResponse.getWarnings()` doesn't return null before I call it here. Thoughts?

Reply:
I've noted that you flipped the condition. Do you think that what you've suggested provides more clarity than
```
if (response.getQueryResponse != null && response.getWarnings != null) {
    response.getQueryResponse()
      .getProcessingDetails()
      .stream()
      // ... etc
      .filter(Objects::nonNull)
      .forEach(response.getWarnings()::addAll);
}
```
within the private method?

Reply:
@kentmorrissey Yes but in this particular case it's not obvious. I prefer structuring my `IF`s by consistently detecting bad cases and handling those eagerly. It'll make for a consistent codebase because it works for the trivial ☝️ and scales to the complex as well. 

A lot of it does come down to user preference, but it does empirically cut down on cyclomatic complexity of code. That's a win IMO.